File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;

class Solution {
    public int solution(int[] elements) {
        int answer = 0;
        Set<Integer> set = new HashSet<>();
        
        int sum=0;
        for(int i=0; i<elements.length; i++){       
            for(int j=0; j<elements.length; j++){
                sum=0;
                for(int k=0; k<=i; k++){        // 길이가 i+1인 부분수열
                    if(j+k >= elements.length){
                        sum += elements[j+k-elements.length];
                    }
                    else{
                        sum += elements[j+k];
                    }
                }
                set.add(sum);
            }
        }
        
        return set.size();
    }
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    answer = 0
    sum_ = 0
    sum_set = set()
    
    for i in range(len(elements)):
        sum_ = elements[i]
        sum_set.add(sum_)
        for j in range(i+1, i+len(elements)):
            sum_ += elements[j%len(elements)] # 원형 수열이므로 배열의 크기를 넘어가면 다시 처음으로
            sum_set.add(sum_)
    return len(sum_set)